components/Dialogs/src/private/MDCAlertControllerView+Private.m

```diff
@@ -0,0 +1,597 @@
+static const CGFloat MDCDialogActionButtonHeight = 36.0;
+static const CGFloat MDCDialogActionButtonMinimumWidth = 48.0;
+
+static const CGFloat MDCDialogMessageOpacity = 0.54f;
+
+@interface MDCAlertControllerView ()
+
+@property(nonatomic, nonnull, strong) UIScrollView *contentScrollView;
+@property(nonatomic, nonnull, strong) UIScrollView *actionsScrollView;
+
+@property(nonatomic, getter=isVerticalActionsLayout) BOOL verticalActionsLayout;
+
+@end
+
+@implementation MDCAlertControllerView {
+    NSMutableArray<MDCFlatButton *> *_actionButtons;
+    BOOL _mdc_adjustsFontForContentSizeCategory;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    self.backgroundColor = [UIColor whiteColor];
+    self.autoresizesSubviews = NO;
+    self.clipsToBounds = YES;
+
+    self.contentScrollView = [[UIScrollView alloc] initWithFrame:CGRectZero];
+    self.contentScrollView.backgroundColor = [UIColor whiteColor];
+    [self addSubview:self.contentScrollView];
+
+    self.actionsScrollView = [[UIScrollView alloc] initWithFrame:CGRectZero];
+    self.actionsScrollView.backgroundColor = [UIColor whiteColor];
+    [self addSubview:self.actionsScrollView];
+
+    self.titleLabel = [[UILabel alloc] initWithFrame:CGRectZero];
+    self.titleLabel.numberOfLines = 0;
+    self.titleLabel.textAlignment = NSTextAlignmentNatural;
+    if (self.mdc_adjustsFontForContentSizeCategory) {
+      self.titleLabel.font = [UIFont mdc_preferredFontForMaterialTextStyle:MDCFontTextStyleTitle];
+    } else {
+      self.titleLabel.font = [MDCTypography titleFont];
+    }
+    [self.contentScrollView addSubview:self.titleLabel];
+
+    self.messageLabel = [[UILabel alloc] initWithFrame:CGRectZero];
+    self.messageLabel.numberOfLines = 0;
+    self.messageLabel.textAlignment = NSTextAlignmentNatural;
+    if (self.mdc_adjustsFontForContentSizeCategory) {
+      self.messageLabel.font = [UIFont mdc_preferredFontForMaterialTextStyle:MDCFontTextStyleBody1];
+    } else {
+      self.messageLabel.font = [MDCTypography body1Font];
+    }
+    self.messageLabel.textColor = [UIColor colorWithWhite:0.0f alpha:MDCDialogMessageOpacity];
+    [self.contentScrollView addSubview:self.messageLabel];
+
+    _actionButtons = [[NSMutableArray alloc] init];
+
+//    CGSize idealSize = [self calculatePreferredContentSizeForBounds:CGRectInfinite.size];
+
+//    self.preferredContentSize = idealSize;
+//    _previousLayoutSize = CGSizeZero;
+
+    [self setNeedsLayout];
+  }
+
+  return self;
+}
+
+- (void)dealloc {
+  [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (NSArray<UIButton *>*)actionButtons{
+  return (NSArray<UIButton *>*)_actionButtons;
+}
+
+- (NSString *)title {
+  return self.titleLabel.text;
+}
+
+- (void)setTitle:(NSString *)title {
+  self.titleLabel.text = title;
+
+//  self.preferredContentSize = [self calculatePreferredContentSizeForBounds:CGRectInfinite.size];
+
+  [self setNeedsLayout];
```

Comment:
do we need this here and in the VC? Can this be called directly?

Reply:
We do not.  Removed from VC.